### DISCUSSION TITLE
Pyinstaller not detecting pandoc on Windows 11

### DISCUSSION THREAD

**Discussion by marph91** [pyinstaller/pyinstaller#8839] [Help] [answered]
Hi, I got an [application built](https://github.com/marph91/jimmy/releases/tag/v0.0.28) and working with Windows 10. When trying to run it on Windows 11, pandoc isn't detected:

```
>jimmy-cli-windows .
Traceback (most recent call last):
  File "jimmy_cli.py", line 130, in <module>
  File "jimmy_cli.py", line 126, in main
  File "jimmy.py", line 109, in jimmy
  File "pypandoc\__init__.py", line 641, in get_pandoc_version
  File "pypandoc\__init__.py", line 797, in _ensure_pandoc_path
OSError: No pandoc was found: either install pandoc and add it
to your PATH or or call pypandoc.download_pandoc(...) or
install pypandoc wheels with included pandoc.
[PYI-23644:ERROR] Failed to execute script 'jimmy_cli' due to unhandled exception!
```

Related issue: https://github.com/marph91/jimmy/issues/17#issuecomment-2423183835

Pandoc is [installed at the build pipeline](https://github.com/marph91/jimmy/blob/63bdd979a501bc7853e3eabcfe55d453e54cbe18/.github/workflows/build.yml#L36) and Pypandoc (a pandoc wrapper) is added [here](https://github.com/marph91/jimmy/blob/63bdd979a501bc7853e3eabcfe55d453e54cbe18/jimmy_cli.spec#L6) to pyinstaller.

Any idea whats the error cause and how to fix it?

**Comment by rokm** [comment]
So where does [`python -c 'from pypandoc.pandoc_download import download_pandoc; download_pandoc(version="3.5")'` ](https://github.com/marph91/jimmy/blob/63bdd979a501bc7853e3eabcfe55d453e54cbe18/.github/workflows/build.yml#L36) actually put the downloaded pandoc executable? Because if it is not in the package's directory, then that `collect_data_files('pypandoc')` is not going to pick it up...

**Reply by rokm** [reply, marked as answer]
[This](https://github.com/JessicaTegner/pypandoc/blob/1a078c0b5075938e57bac531ffcc84d91fbff9b5/pypandoc/pandoc_download.py#L22-L26) seems to imply that it is downloaded into `%localappdata%\Pandoc`. So if that working Windows 10 system is not the build system, then the build application likely works only because you have ran pandoc download at some point and have the executable in the expected location.

**Reply by marph91** [reply]
Well, I think I fooled myself. The command installs pandoc in the system and adds it to the path. The CI and my Windows 10 VM executed the command before running the pyinstaller executable (and using pandoc from the system). The Windows 11 VM is clean and doesn't have pandoc in system or in the pyinstaller executable.

I should just use `pypandoc_binary`, which includes the pandoc executable.

Thanks for the fast answer!
